azimuth/modules/model_contracts/text_classification.py

```diff
@@ -35,6 +36,18 @@ class PipelineOutputProtocol(Protocol):
     postprocessor_output: PostProcessingIO
 
 
+@runtime_checkable
+class PipelineOutputProtocolV2(Protocol):
+    """Class containing result of a batch with pre and postprocessing steps"""
+
+    # model output without passing through post-processing stage: texts, logits, probs, preds
+    model_output: PostProcessingIO
+    # output after passing through post-processing: pre-processed texts, logits, probs, preds
+    postprocessor_output: PostProcessingIO
+    preprocessing_steps: List[Dict[str, Union[str, List[str], int]]]
+    postprocessing_steps: List[Dict[str, Union[str, PostProcessingIO, int]]]
```

Comment:
Why not just add the fields to the existing `PipelineOutputProtocol` as `Optional`?
```suggestion
    preprocessing_steps: Optional[List[Dict[str, Union[str, List[str], int]]]]
    postprocessing_steps: Optional[List[Dict[str, Union[str, PostProcessingIO, int]]]]
```